rust/agama-cli/src/config.rs

```diff
@@ -31,41 +32,57 @@ pub enum ConfigAction {
     Load(String),
 }
 
+fn token() -> Option<String> {
+    match auth::jwt() {
+        Ok(token) => return Some(token),
+        Err(_) => match auth::agama_token() {
+            Ok(token) => return Some(token),
+            Err(_) => return None,
+        },
+    }
+}
+
 pub async fn run(subcommand: ConfigCommands, format: Format) -> anyhow::Result<()> {
-    let store = SettingsStore::new(connection().await?).await?;
+    if let Some(token) = token() {
```

Comment:
If you are not interested in a potential error coming from `token()`, you could use `let else` (to avoid indenting the whole function).

```rust
let Some(token) = token() {
  println!("You need to login for generating a valid token");
  return Ok(());
}
```

Having said that, you should not return `Ok(())` when it is actually an error.